### ISSUE TITLE
Nodes running but not picking up RF

### ISSUE THREAD

**Issue by jayybeerc** [RotorHazard/RotorHazard#654]
This is an open issue as a request for tips please.
Firstly, amazing work by the contributors, first class and very much appreciated.
My issue as the title suggests relates to the nodes running (Showing a trace RF value) but not picking up RF transmitted by anyone of my quads.  My trace RSSI is on average 44 and the node address code is correctly set 1 to 4 (As per the excellent docs).
I am am running version 3 of the server plus relevant node code.
The system is so close to working and is extremely feature rich so this node issue is quite frustrating.
Any help, tips, advise, guidance would be very welcome and again many thanks, great work.

**Comment by jayybeerc** [comment]
Outstanding. Great responses, many thanks. Well, I am getting multiple 0xFF so I will now continue and do the hardware SPI mod. My 5808's are new and the dates supposedly have the SPI mod. The sample test code is great and proves what needs to be done. I will post an update as to the result. Thanks again all, fantastic support.

**Comment by martinglass** [comment]
> Outstanding. Great responses, many thanks. Well, I am getting multiple 0xFF so I will now continue and do the hardware SPI mod. My 5808's are new and the dates supposedly have the SPI mod. The sample test code is great and proves what needs to be done. I will post an update as to the result. Thanks again all, fantastic support.

Glad it's helped. 

FYI, pin 7 on the RTC6715 needs to be HIGH for SPI mode, see
[https://github.com/martinglass/Testing-RX5808-module/blob/main/src/RTC6715-RichWave.pdf](https://github.com/martinglass/Testing-RX5808-module/blob/main/src/RTC6715-RichWave.pdf)

**Comment by jayybeerc** [comment]
Update:
All working and very impressive chaps. The nodes do need a good tune but so far so good. I need to test the system with my racing club both outdoor and indoor. A lot of flyers will be very happy. The features and functions are impressive. Your time and dedication to produce such a great project is humbling. I will let you know how further tests go. Many thanks again.

**Comment by jayybeerc** [comment]
Hello once again. Firstly I wish to express mine and others deep gratitude for all the hard work put in by your good selves. I hope and trust you will be pleased to hear that we have since April 2022 used version 3 of the timer up until last week, so a very happy bunch of flyers for the last 1.5 years. We have found that the timer at version 3 has been stable and works very well, this being from 5" outside racing to indoor whoop meets. We simple could not survive without it. So again, many thanks. OK, so as the timer is so popular I have extended it from a 4 ch to 8ch device. We have decided to remain at version 3 (I will upgrade in the future) as for us it works well, we just need more capacity. ENQUIRY: I read in the version 4 docs that the Arduino node code remains unchanged over the past versions, so have flashed my new 4 nodes, changing the ID number in the header file starting at 5 through to 8. However now when I run the timer I am getting errors in the logs, for example: 

2024-01-03 20:36:53.440: __main__ [INFO] Release: 3.0.0 / Server API: 30 / Latest Node API: 35
2024-01-03 20:36:53.441: RHUtils [INFO] Host machine: Raspberry Pi 4 Model B Rev 1.4
2024-01-03 20:36:53.441: RHUtils [INFO] Host OS: Linux 5.10.60-v7l+
2024-01-03 20:36:53.442: RHUtils [INFO] Python version: 3.7.3 (default, Jan 22 2021, 20:04:44)
2024-01-03 20:36:53.442: __main__ [INFO] Host machine is 'raspberrypi' at 192.168.1.101
2024-01-03 20:36:53.443: Config [INFO] Using configuration file 'config.json'
2024-01-03 20:36:53.443: __main__ [INFO] Using log file: logs/rh_20240103_203653.log
2024-01-03 20:36:53.596: Plugins [INFO] Loaded module i2c_helper
2024-01-03 20:36:53.700: Plugins [INFO] Loaded module i2c_node
2024-01-03 20:36:53.704: Plugins [INFO] Loaded module serial_node
2024-01-03 20:36:53.706: i2c_node [INFO] Searching for I2C nodes...
2024-01-03 20:36:53.707: i2c_node [INFO] ...No I2C node at address 8
2024-01-03 20:36:53.724: i2c_node [INFO] ...No I2C node at address 10
2024-01-03 20:36:53.740: i2c_node [INFO] ...No I2C node at address 12
2024-01-03 20:36:53.757: i2c_node [INFO] ...No I2C node at address 14
2024-01-03 20:36:53.865: i2c_node [INFO] ...I2C node 5 found at address 16, API_level=35, fw_version=1.1.4, fw_type=Arduino, fw_timestamp: Jan  3 2024 13:37:41
2024-01-03 20:36:53.972: i2c_node [INFO] ...I2C node 6 found at address 18, API_level=35, fw_version=1.1.4, fw_type=Arduino, fw_timestamp: Jan  3 2024 13:41:28
2024-01-03 20:36:54.078: i2c_node [INFO] ...I2C node 7 found at address 20, API_level=35, fw_version=1.1.4, fw_type=Arduino, fw_timestamp: Jan  3 2024 13:43:55
2024-01-03 20:36:54.185: i2c_node [INFO] ...I2C node 8 found at address 22, API_level=35, fw_version=1.1.4, fw_type=Arduino, fw_timestamp: Jan  3 2024 13:44:38
2024-01-03 20:36:54.598: __main__ [INFO] Number of nodes found: 4
2024-01-03 20:36:55.199: Plugins [INFO] Loaded module bme280_sensor
2024-01-03 20:36:55.218: Plugins [INFO] Loaded module ina219_sensor
2024-01-03 20:36:55.221: Plugins [INFO] Loaded module linux_sensor
2024-01-03 20:36:55.246: linux_sensor [INFO] Core temperature available
2024-01-03 20:36:55.605: __main__ [INFO] Set Profile to '2'
2024-01-03 20:36:57.752: Plugins [INFO] Loaded module data_export_csv
2024-01-03 20:36:57.756: Plugins [INFO] Loaded module data_export_json
2024-01-03 20:36:57.915: __main__ [INFO] Running http server at port 5000
2024-01-03 20:36:59.231: BaseHardwareInterface [INFO] Interface: Starting background thread
2024-01-03 20:36:59.258: RHInterface [ERROR] Exception in RHInterface update_loop():
Traceback (most recent call last):
  File "../interface/RHInterface.py", line 198, in update_loop
    self.update()
  File "../interface/RHInterface.py", line 347, in update
    data_logger = self.data_loggers[node.index]
IndexError: list index out of range
2024-01-03 20:37:00.285: RHInterface [ERROR] Exception in RHInterface update_loop():
Traceback (most recent call last):
  File "../interface/RHInterface.py", line 198, in update_loop
    self.update()
  File "../interface/RHInterface.py", line 347, in update
    data_logger = self.data_loggers[node.index]
IndexError: list index out of range

Any insight or help very much appreciated please.
Overall we just want to keep what we have and add the extra 4 channels.
In advance, thanks.
JB

**Comment by HazardCreative** [comment]
The timer you have booted here does not have nodes 1–4, only 5–8. The Arduino-based setup requires nodes be available in order, starting from 1.

**Comment by ethomas997** [comment]
That hardware is pretty old so we don't have good documentation on it; there's really just this one page:

https://github.com/RotorHazard/RotorHazard/blob/main/doc/Timer%20Build%20with%20Arduinos.md

If you parallel connect the lines for GND, SDA and SCL between the boards then I think it should work.  Each board will need power as well.

**Comment by HazardCreative** [comment]
> OK I have 2 Delta5 boards (4ch per board) and it is not possible to run them in isolation, so this can be ruled out. 

Currently that is correct. We may be able to fix the error you're getting. Would you isolate it into a new issue, please?

> Do you have a wiring schematic showing the required links between the boards?

No; the Delta5 board predates the RotorHazard project and the schematics weren't included. Your best bet with it is to start the way ET suggested and check with a multimeter that the connections match the [required wiring](https://github.com/RotorHazard/RotorHazard/blob/main/doc/Timer%20Build%20with%20Arduinos.md#system-assembly). You could also migrate your components over to the [RH Arduino PCB](https://github.com/RotorHazard/RotorHazard/blob/main/resources/PCB/README.md), which has a clearly labeled header to connect between two PCBs.

I'm *specifically not recommending this* as I don't think it's worth the risk or effort, but the RH PCB can auto-number nodes at startup. If you put a switch across JP2 on PCB2, you could create a system that numbers itself from either 1-4 or 5-8 based on the position of that switch at boot.

**Comment by jayybeerc** [comment]
> Currently that is correct. We may be able to fix the error you're getting. Would you isolate it into a new issue, please?
No need I have worked around the problem.

I must remark and express my thanks for the replies and your valued time.
I now have a working timer that has migrated from 4 to 8 ch's.
I know and appreciate that I have been working with older software revisions and hardware but the object was to take a 4ch device and simple extend to 8 as the original system was and still is fit for purpose. 
I am not sure if anyone is interested in what was required to get it up and running but after a few hundred lap tests on nodes 5 to 8 (I only have 1 extra RX5808 chip that I moved to each channel) all seem OK.
So for now the plan is to run with this version and migrate the hardware and upgrade the software to the latest when convenient.
Finally, thanks again for all the replies and this amazing project.
Much appreciated.
